src/main/java/com/arpnetworking/metrics/mad/model/json/Version2g.java

```diff
@@ -0,0 +1,394 @@
+            _gauges = value;
+            return this;
+        }
+
+        @NotNull
+        private String _id;
+        @NotNull
+        private DateTime _start;
+        @NotNull
+        private DateTime _end;
+        @NotNull
+        private ImmutableMap<String, String> _dimensions;
+        @NotNull
+        private ImmutableMap<String, String> _annotations;
+        private ImmutableMap<String, Element> _counters;
+        private ImmutableMap<String, Element> _gauges;
+        private ImmutableMap<String, Element> _timers;
+        @NotNull
+        @MatchPattern(pattern = "^2g$", flags = Pattern.CASE_INSENSITIVE)
+        private String _version;
+    }
+
+    /**
+     * Represents a single sample.
+     */
+    public static final class Sample {
+
+        public Unit getUnit2g() {
+            return _unit;
+        }
+
+        public double getValue() {
+            return _value;
+        }
+
+        private Sample(final Builder builder) {
+            _unit = builder._unit;
+            _value = builder._value;
+        }
+
+        private final Unit _unit;
+        private final double _value;
+
+        /**
+         * Builder for the Sample class.
+         */
+        public static final class Builder extends OvalBuilder<Sample> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(Sample::new);
+            }
+
+            /**
+             * Sets the value field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setValue(final Double value) {
+                _value = value;
+                return this;
+            }
+
+            /**
+             * Sets the compound unit field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setUnit(final Unit value) {
+                _unit = value;
+                return this;
+            }
+
+            @NotNull
+            private Double _value;
+            private Unit _unit;
+        }
+
+        /**
+         * 2G format Unit class.
+         */
+        public static final class Unit {
+
+            public List<CompositeUnit> getNumerators() {
+                return _numerators;
+            }
+
+            public List<CompositeUnit> getDenominators() {
+                return _denominators;
+            }
+
+            private Unit(final Builder builder) {
+                _numerators = builder._numerators == null
+                        ? ImmutableList.of()
+                        : ImmutableList.copyOf(builder._numerators);
+                _denominators = builder._denominators == null
+                        ? ImmutableList.of()
+                        : ImmutableList.copyOf(builder._denominators);
+            }
+
+            private final List<CompositeUnit> _numerators;
```

Comment:
These should be stored and returned as ImmutableList so callers don't need to defensively copy.